### DISCUSSION TITLE
Question about HostKeyVerification

### DISCUSSION THREAD

**Discussion by hamnis** [hierynomus/sshj#839] [Q&A] [answered]
In 
https://github.com/hierynomus/sshj/blob/master/src/main/java/net/schmizz/sshj/transport/kex/AbstractDHG.java#L99-L127

There is code to verify the HostKey. Why is this needed when you have `HostKeyVerification` instances registered in `KeyExchanger`?

What happens if we turn off the trans.getConfig().isVerifyHostKeyCertificates(), do we also turn off the HostKeyVerifications registered in 
KeyExchanger?

The server I am connecting to is using VerifyHostKeyDNS and this does not seem supported.

Thanks for any response.

**Comment by hierynomus** [comment, marked as answer]
No, this does not verify the host key itself (which could be a certificate!), but rather whether we trust the signature on the  certificate if the host key is a certificate.

It could be for instance that you you've added the (certificate) host key to your `known_hosts`, but at a certain moment in time the certificate becomes revoked, the signing CA gets breached and retracts the root certificate, etc. It adds a layer of security.

The `HostKeyVerifier` just checks whether the hostkey is known, by checking a hardcoded/configured fingerprint, looking at your `known_hosts` file, or just blindly accepting all.

**Reply by hamnis** [reply]
Ok. thanks for the reply.

Is there a way to use VerifyHostKeyDNS with this library?

On the command line this works even though the server `sftp.example.com` is located at `someotherhost.example.com`
https://man.openbsd.org/ssh_config#VerifyHostKeyDNS

**Reply by hierynomus** [reply]
Currently this is not available. quickly glancing through the docs for sshfp, it should be possible to add though.
